tls/s2n_prf.c

```diff
@@ -408,6 +409,31 @@ int s2n_hybrid_prf_master_secret(struct s2n_connection *conn, struct s2n_blob *p
     return s2n_prf(conn, premaster_secret, &label, &client_random, &server_random, &conn->kex_params.client_key_exchange_message, &master_secret);
 }
 
+int s2n_prf_calculate_master_secret(struct s2n_connection *conn, struct s2n_blob *premaster_secret)
+{
+    POSIX_ENSURE_REF(conn);
+
+    if (conn->ems_negotiated && S2N_IN_TEST) {
```

Comment:
This method now MUST be called from the client key message, or it will fail. That seems uncomfortably brittle, but we probably need to minimally enforce it's actually true. Like make sure the current message is the client key message.

Reply:
Or at least that the client is the writer for the current message.